desc/perturbations.py

```diff
@@ -664,7 +618,6 @@ def optimal_perturb(  # noqa: C901 - FIXME: break this up into simpler pieces
             print("Computing dg")
```

Comment:
Is dg also needed when I only have one objective like "force" or "current"? It thought dg was a set of objectives other than the force error df

Reply:
Yes, there are two different forms of a perturbation. In the first, you fix ahead of time what the changes in the parameters $\Delta c$ are, in which case you only have a single objective (usually force or current). This is what's done in `perturb`

The other one we choose $\Delta c$ to minimize a second objective `g`. This is what's done in `optimal_perturb`. It's basically one step of a constrained optimization loop.